restcomm/restcomm.ussd/src/main/java/org/restcomm/connect/ussd/interpreter/UssdInterpreter.java

```diff
@@ -545,27 +545,34 @@ else if ("outbound-api".equals(response.get().direction())){
             }
         } else if (DownloaderResponse.class.equals(klass)) {
             final DownloaderResponse response = (DownloaderResponse) message;
-
-            int sc = response.get().getStatusCode();
-            //processingInfoRequest
             if (logger.isDebugEnabled()) {
-                logger.debug("Rcml URI : " + response.get().getURI() + "response success=" + response.succeeded()
-                    + ", statusCode=" + response.get().getStatusCode()+" state="+state);
+                logger.debug("Rcml DownloaderResponse success=" + response.succeeded());
             }
             //FIXME: what if these ifblocks arent in downloadingRcml?
             if (response.succeeded()) {
+                int sc = response.get().getStatusCode();
+                //processingInfoRequest
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Rcml DownloaderResponse URI : " + response.get().getURI()
+                        + ", statusCode=" + response.get().getStatusCode()+" state="+state);
+                }
                 if(HttpStatus.SC_OK == sc){
                     fsm.transition(message, ready);
                 } else if (HttpStatus.SC_NOT_FOUND == sc){
                     fsm.transition(message, notFound);
                 }
             } else {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Rcml DownloaderResponse error : " + response.error()
+                        + ", cause=" + response.cause().getMessage());
```

Comment:
im not sure "cuase()" will return an expcetion always in this branch, it may potentially throw NPE...

Reply:
@jaimecasero 
Seems like theres only two places where `DownloaderResponse` is instantiated, so it doesnt look like anyone every passes a null to exception.
https://github.com/RestComm/Restcomm-Connect/blob/007020dc54fa069ef68ab01424bf17a049cebd9a/restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/client/Downloader.java#L187

Reply:
im talkign abotu this expression "response.cause().getMessage()", and particularly how "response.cause()" may lead to null, not "response" object itself. I just dont want to chagne an NPE for other...

Reply:
What i pointed out was there does not seem to be any other Download failure instance where we instantiate the `DownloaderResponse` with null `exception` instead.
Line 187 is the only `DownloaderResponse` failure instance, so when it fails `exception` is always set. Hence there is no failure case where `DownloaderResponse.cause()` is null.

Reply:
Made an inline null check, dont really like that idiom for sure, but maybe easiest compromise. Advise if you think we should do away with it.